### ISSUE TITLE
Failing test(s):  TestAccGkeonprem* (beta)

### ISSUE THREAD

**Issue by SarahFrench** [hashicorp/terraform-provider-google#14417]
### Failure rates

- 100% since 2023-04-02 (start of test being run in CI)

### Impacted tests

- TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
- TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterF5lbExample
- TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterManuallbExample
- TestAccGkeonpremVmwareCluster_vmwareClusterUpdateBasic
- TestAccGkeonpremVmwareCluster_vmwareClusterUpdateF5Lb
- TestAccGkeonpremVmwareCluster_vmwareClusterUpdateManualLb
- TestAccGkeonpremBareMetalCluster_gkeonpremBareMetalClusterBasicExample
- TestAccGkeonpremBareMetalCluster_gkeonpremBareMetalClusterManuallbExample
- TestAccGkeonpremBareMetalCluster_gkeonpremBareMetalClusterBgplbExample
- TestAccGkeonpremBareMetalCluster_bareMetalClusterUpdateBasic
- TestAccGkeonpremBareMetalCluster_bareMetalClusterUpdateManualLb
- TestAccGkeonpremBareMetalCluster_bareMetalClusterUpdateBgpLb
- TestAccGkeonpremBareMetalNodePool_gkeonpremBareMetalNodePoolBasicExample
- TestAccGkeonpremBareMetalNodePool_gkeonpremBareMetalNodePoolFullExample
- TestAccGkeonpremBareMetalNodePool_bareMetalNodePoolUpdate
- TestAccGkeonpremVmwareNodePool_gkeonpremVmwareNodePoolBasicExample
- TestAccGkeonpremVmwareNodePool_gkeonpremVmwareNodePoolFullExample
- TestAccGkeonpremVmwareNodePool_vmwareNodePoolUpdate

### Affected Resource(s)

* google_gkeonprem_vmware_cluster

### Nightly build test history

- [Link](https://ci-oss.hashicorp.engineering/test/6958106216713475915?currentProjectId=GoogleCloudBeta&branch=%3Cdefault%3E)

### Message(s)

Original error, has been solved:
```
Error: Error creating VmwareCluster: googleapi: Error 403: GKE On-Prem API has not been used in project 1234567890 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/gkeonprem.googleapis.com/overview?project=1234567890 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
```

Newer error:

```
------- Stdout: -------
=== RUN   TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterF5lbExample
=== PAUSE TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterF5lbExample
=== CONT  TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterF5lbExample
vcr_utils.go:146: Step 1/2 error: Error running apply: exit status 1
Error: Error creating VmwareCluster: googleapi: Error 403: permission crossing-project "1234567890" denied on "projects/0987654321/locations/global/memberships/gkeonprem-terraform-test"
Details:
[
{
"@type": "type.googleapis.com/google.rpc.RequestInfo",
"requestId": "511e777549cf09bf"
}
]
with google_gkeonprem_vmware_cluster.cluster-f5lb,
on terraform_plugin_test.tf line 2, in resource "google_gkeonprem_vmware_cluster" "cluster-f5lb":
2: resource "google_gkeonprem_vmware_cluster" "cluster-f5lb" {
--- FAIL: TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterF5lbExample (3.12s)
FAIL
```

**Comment by SarahFrench** [comment]
This new resource and their tests are only in the Beta provider currently, but the API could be activated in both GA and Beta nightly test projects at the same time in anticipation of this going GA

**Comment by SarahFrench** [comment]
This API was activated yesterday

**Comment by SarahFrench** [comment]
I've updated the issue description as a new error is showing now, following activation of the API.

**Comment by trodge** [comment]
b/280098984

**Comment by naitianliu-google** [comment]
Could you please help us understand what the crossing-project test is for? Does it update the project only?

**Comment by roaks3** [comment]
@naitianliu-google  Could you clarify the question? Are you asking about the intent of the `TestAccGkeonpremVmwareCluster_gkeonpremVmwareClusterF5lbExample` test?

For a little more context, the project that is being referenced as the "crossing-project" in this error message is the main GCP project where we run our nightly test suite from. It seems that we need to grant permission for that project to access the memberships on `870316890899` (which appears to be another internal test project). Perhaps this was already done for the project that runs our PR tests?

**Comment by megan07** [comment]
The [`TestAccGkeonpremBareMetal*` tests](https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaGoogleProject/398949?mode=builds&buildTab=tests&status=failed&pager.currentPage=1&expandedTest=9122160593572963491) seem to be failing for the same reason as of May 5/6, would this fall under the scope of this issue or should I create a different one?

**Comment by roaks3** [comment]
Yea, I would include it here. I believe the service team is adding new resources/tests using this same setup, and it doesn't look like the access issue has been resolved during that time.

I'm also going to ping the internal ticket.

**Comment by roaks3** [comment]
FYI I've skipped the tests that have been listed here so far

**Comment by naitianliu-google** [comment]
Thanks for checking this. I have added permission for the nightly projects in API backends. However, the project info in the test sample should be dynamically changed to current project as well. https://github.com/naitianliu-google/magic-modules/blob/5210faadb783503fdfdf9dfa796b98c222ad1db9/mmv1/templates/terraform/examples/gkeonprem_vmware_cluster_basic.tf.erb#L5

Could you please help guide if there is way to dynamically substitute the project in test examples? CC: @roaks3

**Comment by naitianliu-google** [comment]
Thanks for the guidance! Yes, we need to supply a different project number in the test depending on which nightly project is currently in use. This is because we have validation logic rejecting cross-project requests on server side. Let me also check with my team to see we can work around the validation on server side.

**Comment by naitianliu-google** [comment]
We have bypassed validations for the nightly projects: 
`ci-test-project-nightly-beta`
`ci-test-project-nightly-ga`

Could you please help enable the gkeonprem API for the projects? @roaks3 Thanks!
```
name: Anthos On-Prem API
url: https://console.cloud.google.com/apis/library/gkeonprem.googleapis.com
```

**Comment by roaks3** [comment]
Ah ok, sounds good! It looks like that API has already been enabled for those projects, so there should be a clear path once your PR is merged.

**Comment by SarahFrench** [comment]
It looks like the permission issues are resolved. but there's a problem with the acceptance tests creating a cluster with the same name, and possibly that cluster not being cleaned up between test runs. I've put the details in a new issue, as _this_ issue around permissions and API activation.seems to be resolved.

New issue here : https://github.com/hashicorp/terraform-provider-google/issues/15094

**Comment by roaks3** [comment]
Yep, permission issue appears to be resolved
